Neos.Media/Classes/Domain/Service/AssetService.php

```diff
@@ -308,6 +308,8 @@ public function replaceAssetResource(AssetInterface $asset, PersistentResource $
                             $newVariantResourceUri = new Uri($this->resourceManager->getPublicPersistentResourceUri($variant->getResource()));
                             $uriMapping[$originalVariantResourceUri->getPath()] = $newVariantResourceUri->getPath();
                         }
+
+                        $this->getRepository($variant)->add($variant);
```

Comment:
Seem not needed, at least not with latest 5.1. The recreated variant is added to the image and thus persisted with `Image` through the `$variants` property that has `cascade={"all"}` set.